src/aoefit.jl

```diff
@@ -327,3 +327,183 @@ function fit_single_aoe_compton(h::Histogram, ps::NamedTuple; uncertainty::Bool=
+        f_sig = x -> Base.Fix2(f_aoe_sig, v_ml)(x),
+        f_bck = x -> Base.Fix2(f_aoe_bkg, v_ml)(x)
+    )
+    
+    return Optim.minimum(opt_r), report
+end
+
+"""
+    fit_aoe_compton_combined(peakhists::Vector{<:Histogram}, peakstats::StructArray, compton_bands::Array{T}, result_corrections::NamedTuple; pars_aoe::NamedTuple{(:μ, :μ_err, :σ, :σ_err)}=NamedTuple{(:μ, :μ_err, :σ, :σ_err)}(nothing, nothing, nothing, nothing), uncertainty::Bool=false) where T<:Unitful.Energy{<:Real}
+
+Performed a combined fit over all A/E Compton band using the `f_aoe_compton` function consisting of a gaussian SSE peak and a step like background for MSE events,
+assuming `f_aoe_mu` for μ and `f_aoe_sigma` for σ.
+
+# Returns
+    * `v_ml`: The fit result from the maximum-likelihood fit.
+    * `report`: Dict of NamedTuples of the fit report which can be plotted for each compton band
+"""
+function fit_aoe_compton_combined(peakhists::Vector{<:Histogram}, peakstats::StructArray, compton_bands::Array{T}, result_corrections::NamedTuple; pars_aoe::NamedTuple{(:μ, :μ_err, :σ, :σ_err)}=NamedTuple{(:μ, :μ_err, :σ, :σ_err)}(nothing, nothing, nothing, nothing), uncertainty::Bool=false) where T<:Unitful.Energy{<:Real}
+    
+    μA = ustrip(result_corrections.μ_compton.par[1])
+    μB = ustrip(result_corrections.μ_compton.par[2])
+    σA = ustrip(result_corrections.σ_compton.par[1])
+    σB = ustrip(result_corrections.σ_compton.par[2])
+    
+    # create pseudo priors
+     pseudo_prior = NamedTupleDist(
+        μA = Normal(mvalue(μA), muncert(μA)),
+        μB = Normal(mvalue(μB), muncert(μB)),
+        σA = Normal(mvalue(σA), muncert(σA)),
+        σB = Normal(mvalue(σB), muncert(σB)),
+    )
+    
+    # transform back to frequency space
+    f_trafo = BAT.DistributionTransform(Normal, pseudo_prior)
+    
+    # start values for MLE
+    v_init = mean(pseudo_prior)
+
+    # create loglikehood function
+    f_loglike = pars -> begin
+        
+        neg_log_likelihoods = zeros(typeof(pars.μA), length(compton_bands))
+
+        # iterate throuh all peaks (multithreaded)
+        Threads.@threads for i in eachindex(compton_bands)
+
+            # get histogram and peakstats
+            h  = peakhists[i]
+            ps = peakstats[i]
+            e = ustrip(compton_bands[i])
+            μ = f_aoe_mu(e, (pars.μA, pars.μB))
+            σ = f_aoe_sigma(e, (pars.σA, pars.σB))
+
+            # fit peak
+            try
+                A, B = fit_single_aoe_compton_with_fixed_μ_and_σ(h, μ, σ, ps; uncertainty=uncertainty)
+                neg_log_likelihoods[i] = A
+            catch e
+                @warn "Error fitting band $(compton_bands[i]): $e"
+                continue
+            end
+        end
+        return sum(neg_log_likelihoods)
+    end
+    
+    # MLE
+    opt_r = optimize(f_loglike ∘ inverse(f_trafo), f_trafo(v_init), NelderMead(), Optim.Options(time_limit = 60, show_trace=false, iterations = 1000))
```

Comment:
Changed the number of iterations (still needs to be properly tested, these are arbitrary numbers)